charms/argo-controller/terraform/main.tf

```diff
@@ -0,0 +1,13 @@
+resource "juju_application" "argo_controller" {
+  name      = var.app_name
+  model     = var.model_name
```

Comment:
This reference will happen from a higher level module. 
e.g. in a higher level `main.tf` file, there will be
```
resource "juju_model" "kubeflow" {
  name = var.model_name
}

module "argo-controller" {
  source = "<argo-controller-dricetory>"
  model_name = juju_model.kubeflow.name
}
```

Doesn't this ensure that Terraform is aware of the dependency between `module.argo_controller` and `juju_model.kubeflow`?

Reply:
So from our discussion, the problem is that defining the input `model_name` as a string and using it directly in the `juju_application` resource doesn't expose the requirement of the charm that it needs a `juju_model` to be available with the same name, and thus "leads users down the wrong path - one we know will cause problems" as per Heather's words. This can result in the scenarions where the end user:
* tries to use the charm module in a solution of theirs, without defining a `juju_model` and getting a `model not found` error
* defining a `juju_model` resource but instead of passing the `juju_model.<model_label>.name` as an input to the module's `model_name` input, they 're passing the  `<model-name>` string. This results in Terraform not being aware of the dependency and since Terraform is creating resources in parallel, this could also result in the above `mode not found` error too.

Thus, we need to find a way to expose this requirement in the module itself.

Reply:
Since this will be a discussion going on in the spec, I 've pushed a README exposing this requirement so we can move forward for now. Let me know what you think.